### PR TITLE
[Winlogbeat] Add missing query while reading .evtx file

### PR DESCRIPTION
**Pull request by kcreddy**
## What does this PR do?
Adds query filter when processing .evtx file

## Why is it important?

- Fixes the bug when reading .evtx files

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.
- [x] I have made my commit title and message explanatory about the purpose and the reason of the change

## Author's Checklist

- [ ]

## Related issues

- Closes https://github.com/elastic/beats/issues/16826

### CODE REVIEW

File: winlogbeat/eventlog/wineventlog_experimental.go [hunk 1]
```diff
@@ -256,7 +256,7 @@ func (l *winEventLogExp) openChannel(bookmark win.Bookmark) (win.EvtHandle, erro
 }
 
 func (l *winEventLogExp) Read() ([]Record, error) {
-	var records []Record
+	records := make([]Record, 0, l.maxRead)
```
Comment (andrewkroh) [inline winlogbeat/eventlog/wineventlog_experimental.go:259]: How is this modification related to the bugfix? Same question for L427 in 
winlogbeat/eventlog/wineventlog.go.

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -33,6 +33,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 *Winlogbeat*
 
 - Add "event.category" and "event.type" to Sysmon module for EventIDs 8, 9, 19, 20, 27, 28, 255 {pull}35193[35193]
+- Add missing query while reading ".evtx" file. {issue}16826[16826] {pull}36173[36173]
```
Comment (andrewkroh) [inline CHANGELOG.next.asciidoc:36]:
```suggestion
- Fix the ability to use filtering features (e.g. `ignore_older`, `event_id`, `provider`, `level`) while reading `.evtx` files. {issue}16826[16826] {pull}36173[36173]
```

File: winlogbeat/eventlog/wineventlog_experimental.go (hunk 1 above)
Comment (kcreddy) [reply winlogbeat/eventlog/wineventlog_experimental.go:259]: This was to make `golangci-lint` to pass. 
This was the error: `golangci lint error: consider pre-allocating records`. In both places.

File: CHANGELOG.next.asciidoc (hunk 2 above)
Comment (kcreddy) [reply CHANGELOG.next.asciidoc]: Updated the changelog

File: winlogbeat/eventlog/wineventlog_experimental.go (hunk 1 above)
Comment (kcreddy) [reply winlogbeat/eventlog/wineventlog_experimental.go]: Thanks Andrew for the details. I added the `//nolint` directive.

**Review by andrewkroh — APPROVED** [review]
